src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table-key-value/table-key-value.component.ts

```diff
@@ -188,6 +189,21 @@ export class TableKeyValueComponent implements OnInit, OnChanges {
     } else if (isEmpty && !this.hideEmpty && v.value !== '') {
       v.value = '';
     }
+    if (!isEmpty && _.isString(v.value)) {
+      v.value = this.convertString(v.value);
+    }
     return v;
   }
+
+  private convertString(s: string) {
+    const date = this.isDate(s) && this.datePipe.transform(s);
```

Comment:
Since we already use moment in other places, I would suggest we use its validation:
```suggestion
    const date = moment(s).isValid() && this.datePipe.transform(s);
```

Reply:
Tested it and it didn't work :/
You can specify a format but that would be to strict as there are different date formats in use. Moment does not have a validation that is as flexible but yet as strict as needed like `isDate`.

Without specifying a format `1, 2, 3` is though as a valid date because it can be transformed into one by `Date` ...

I also tested how it would look like if I specified a format - but that didn't work, `1, 2, 3` kept to be valid...

I'm in favor of not using the moment validation as it seems not to be suited for this purpose.